server/src/main/java/keywhiz/service/resources/automation/v2/BackupResource.java

```diff
@@ -0,0 +1,153 @@
+import keywhiz.api.model.Group;
+import keywhiz.log.AuditLog;
+import keywhiz.log.Event;
+import keywhiz.service.config.Readonly;
+import keywhiz.service.daos.GroupDAO;
+import keywhiz.service.daos.GroupDAO.GroupDAOFactory;
+import keywhiz.service.daos.SecretController;
+import org.bouncycastle.openpgp.PGPException;
+import org.c02e.jpgpj.Encryptor;
+import org.c02e.jpgpj.FileMetadata;
+import org.c02e.jpgpj.Key;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.time.Instant.now;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+import static javax.ws.rs.core.MediaType.APPLICATION_OCTET_STREAM;
+import static keywhiz.log.EventTag.GROUP_BACKUP;
+import static org.c02e.jpgpj.CompressionAlgorithm.ZIP;
+import static org.c02e.jpgpj.EncryptionAlgorithm.AES256;
+import static org.c02e.jpgpj.FileMetadata.Format.UTF8;
+import static org.c02e.jpgpj.HashingAlgorithm.Unsigned;
+
+/**
+ * @parentEndpointName automation/v2-backup
+ * @resourceDescription Automation endpoints to manage groups
+ */
+@Path("/automation/v2/backup")
+public class BackupResource {
+  private static final Logger logger = LoggerFactory.getLogger(BackupResource.class);
+
+  private final GroupDAO groupDAO;
+  private final SecretController secretController;
+  private final ObjectMapper objectMapper;
+  private final KeywhizConfig config;
+  private final AuditLog auditLog;
+
+  @Inject
+  public BackupResource(
+      GroupDAOFactory groupDAOFactory,
+      @Readonly SecretController secretController,
+      ObjectMapper objectMapper,
+      KeywhizConfig config,
+      AuditLog auditLog) {
+    this.groupDAO = groupDAOFactory.readonly();
+    this.secretController = secretController;
+    this.objectMapper = objectMapper;
+    this.config = config;
+    this.auditLog = auditLog;
+  }
+
+  /**
+   * Backup all secrets for a given group. Returns an encrypted encrypted to
+   * the backup key in the main configuration file. Only accessible to automation clients.
+   *
+   * @excludeParams automationClient
+   * @param name Group name
+   *
+   * @return Encrypted archive
+   */
+  @Timed @ExceptionMetered
+  @GET @Path("{group}")
+  @Produces(APPLICATION_OCTET_STREAM)
+  public byte[] backup(
+      @Auth AutomationClient automationClient,
+      @PathParam("group") String name) {
+    Optional<Group> groupOptional = groupDAO.getGroup(name);
+    if (!groupOptional.isPresent()) {
+      throw new NotFoundException("Unknown group: " + name);
+    }
+
+    if (config.getBackupExportKey() == null) {
+      throw new ServiceUnavailableException("Backups keys are not configured");
+    }
+
+    Group group = groupOptional.get();
+
+    // SecretDeliveryResponse is the same data a client receives when requesting a secret,
+    // so it should have all the relevant information we need (including content, checksum).
```

Comment:
This will lose information on the secret's expiration date, as well as some less important fields (such as who created and updated it and any previous versions of the secret)